### PR TITLE
ODC-7648: Open Lightspeed Chat Interface by clicking a button on the YAML Editor component

### PR DESCRIPTION
**Pull request by Lucifergene**
**Type**: 

- [ ] Bug Fix
- [x] New Feature

**Jira Link**: 

1. https://issues.redhat.com/browse/ODC-7648

**Solution Description**: 

- [x] Create the button that will show on top of the YAML Editor
- [x] Add the logic to make it open the Lightspeed Chat Interface once its clicked.

**Screenshots / Videos for review**: 

**Walkthrough**:

https://github.com/user-attachments/assets/e81bf79e-8576-4e82-a049-a55920311f25

**UI**:

image

image

**Test coverage**: 

- No Updates

**Test setup:**

1.  Install the Lightspeed Operator
```
oc apply -f https://raw.githubusercontent.com/openshift-dev-console/operator-subscriptions/main/lightspeed-catalog.yaml
```
2. Go the OperatorHub page and install the Lightspeed Operator.
3. Create the OLSConfig:
```
oc apply -f https://raw.githubusercontent.com/openshift-dev-console/operator-subscriptions/main/ols-config.yaml
```
2. Go to the Import YAML Page
5. Click on the "Ask Openshift Lightspeed" button

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Comment by lokanandaprabhu** [comment]
Hi @Lucifergene ,

I am not able to click the "Ask Openshift Lightspeed" button, anything I am missing here?

https://github.com/user-attachments/assets/b2e0e617-efed-4ed6-9f9c-628a992ba604

**Comment by beaumorley** [comment]
Overall this looks great. Two questions:

1.) What PF component is being used for the Ask OpenShift Lightspeed button?
2.) If you switch to dark mode, does the button invert? 

Also can you change the S in OpenShift in the button to an uppercase S? 
Right now it is "Openshift "and it should be "OpenShift".
Thank you!

**Comment by asuwebdesign** [comment]
To coincide with what @beaumorley said, I only have a minor suggestion:

- Use the `Button` component from [PatternFly](https://www.patternfly.org/components/button#small-buttons) with the `small` and `secondary` variations.

This will allow it to look more consistent with the UI. I believe when OCP updates the PF 6.x, the button should automatically update to look more pill-shaped as you have it visualized. And it should have the proper theming styles too, so it'll maintain contrast in dark mode.

**Comment by Lucifergene** [comment]
@beaumorley Answering your questions:
1. I have currently used the [Label](https://www.patternfly.org/components/label#outlined-labels) component for the button
2. If switched to dark mode, the Purple content stays the same while the border becomes brighter.

Sure, I will change the case in OpenShift.

Also, as @asuwebdesign suggested, I can update the component to Button. I saw we can add icons to Button which would make it similar to how it is now.

**Comment by Lucifergene** [comment]
Final UI Update after the changes proposed by @asuwebdesign & @beaumorley :

image

**Review by lokanandaprabhu — APPROVED** [review]

**Comment by lokanandaprabhu** [comment]
Screenshot 2024-07-30 at 12 55 56 PM
Screenshot 2024-07-30 at 12 56 16 PM

Tested for Admin users. Working fine.

For non-admin users, added the role and tested. Working fine.

/lgtm

**Review by vikram-raj — APPROVED** [review]
/approve

**Comment by sanketpathak** [comment]
As the cluster bot is not working at the moment, I've tested locally with admin and non-admin users. The button looks fine

Screenshot 2024-08-03 at 3 30 13 AM

/label qe-approved

File: frontend/packages/console-shared/src/hooks/ols-hook.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  kind: 'OLSConfig',
+  id: 'olsconfig',
+  labelPlural: 'OLSConfigs',
+  crd: true,
+};
+
+const OLSCONFIG_INSTANCE = 'cluster';
+
+export const useOLSConfig = (): boolean => {
+  const [, loaded, err] = useK8sGet<K8sResourceKind>(OLSConfigModel, OLSCONFIG_INSTANCE);
```
Comment (spadgett) [inline frontend/packages/console-shared/src/hooks/ols-hook.ts:24]: Will normal users be able to get this resource?

**Pull request merged**